### PR TITLE
Substrate v2.0.0

### PR DESCRIPTION
**Pull request by AurevoirXavier**
### **Target**
  - substrate 7314a78d49d0dc3862c6ef7c8115cb1b3d0c0fd2
  - Polkadot 0.8.25

---

### **Remain** 
- [x] #6948
- [x] #6999
- [x] #7058
- [x] #7039
- [x] #5715
- [x] #6215
- [x] #7136
- [x] #7155
- [x] #7161
- [x] #7176
- [x] #6685
- [x] #7168
- [x] #7111
- [x] #7205
- [x] #7214
- [x] #7215

### CODE REVIEW

**Comment by wuminzhe** [comment]
why to polkadot 0.8.25? @AurevoirXavier

**Comment by AurevoirXavier** [comment]
> why to polkadot 0.8.25? @AurevoirXavier

If something wrong. We can follow the polkadot's solution.

**Review by clearloop — APPROVED** [review]
The substrate PR checklist has been completed, CI has been passed.

**Comment by AurevoirXavier** [comment]
@HackFisher Please review this part

File: frame/treasury/src/lib.rs [hunk 1]
```diff
@@ -155,99 +206,154 @@ pub trait Trait: frame_system::Trait {
...
-	type KtonBurnDestination: OnUnbalancedKton<KtonNegativeImbalance<Self>>;
+	type KtonBurnDestination: OnUnbalancedKton<KtonNegativeImbalance<Self, I>>;
 
 	/// Weight information for extrinsics in this pallet.
 	type WeightInfo: WeightInfo;
 }
 
 decl_storage! {
-	trait Store for Module<T: Trait> as DarwiniaTreasury {
+	trait Store for Module<T: Trait<I>, I: Instance = DefaultInstance> as DarwiniaTreasury {
```
Comment (HackFisher) [inline frame/treasury/src/lib.rs:266]: How different Instances allocate their storage keys prefixes?

**Review by HackFisher — APPROVED** [review]

File: frame/treasury/src/lib.rs (hunk 1 above)
Comment (AurevoirXavier) [reply frame/treasury/src/lib.rs:266]: with suffix _InstanceXX, but I didn't see any migration in the origin code

Comment (AurevoirXavier) [reply frame/treasury/src/lib.rs:266]: DefaultInstance is fine

**Pull request merged**